feat: KBOMG-27 - Eerste poging tot vl-click-action component

wimdetroyer opened the pull request:
- introductie vl-click-action-component
- verbeteren typering bij map

kspeltix committed:
feat: optimalisatie mbt github-actions caching

als key wordt nu de package-lock.json gebruikt, hierdoor kan de cache herbruikt worden over builds heen

wimdetroyer committed:
feat: KBOMG-27 - Eerste poging tot vl-click-action component

- introductie vl-click-action-component
- verbeteren typering bij map

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action-pindrop.ts:
```diff
@@ -0,0 +1,68 @@
+export class VlMapClickActionPindrop extends LitElement {
+
+
+    // TODO: huisstijl vlaanderen?
```
Graag commentaar

wimdetroyer replied:
te bekijken... @kspeltix

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action-pindrop.ts:
```diff
@@ -0,0 +1,68 @@
+
+
+    // TODO: huisstijl vlaanderen?
+    // TODO: storybook element hiervoor nodig?
```
Graag commentaar

wimdetroyer replied:
voor 't moment hier geen storybook element voor aangemaakt aangezien het enkel bestaat i/d context van clickaction @kspeltix

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.stories-doc.mdx:
```diff
@@ -0,0 +1,18 @@
+
+## Voorbeeld
+
+TODO: check
```
TODO

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.ts:
```diff
@@ -0,0 +1,78 @@
+}
+
+
+//TODO: interactie met controls? (zoals current location)
```
TODO

wimdetroyer replied:
werkt

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.ts:
```diff
@@ -0,0 +1,78 @@
+
+
+//TODO: interactie met controls? (zoals current location)
+//TODO: wat als geklikt wordt op zelfde locatie van een feature?
```
TODO

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.ts:
```diff
@@ -0,0 +1,78 @@
+ * @property {function} onClick - callback die afgevuurd wordt na click op map
+ */
+
+// TODO: crs doorsturen? vb: crs=EPSG:31370 => komt uit map??
```
Weet niet of dit nu effectief nodig is... projectie is map gerelateerd? @driesva

driesva replied:
ik denk dat het aan de map experten (team Decibel?) is om hier een beter antwoord op te geven. Voorlopig is vl-map altijd Lamber72 dacht ik, dus nog niet nodig.
Mijn comment was eerder toekomstgericht.

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.ts:
```diff
@@ -0,0 +1,78 @@
+
+
+    /**
+     * TODO: open tot discussie
```
Te bediscussieren

wimdetroyer replied:
Ik ga het omzetten naar events. Dat is vrij simpel te bekomen

driesva replied:
ik ben voorstander om `@event` cfr Lit style te gaan gebruiken. Maar zoals gezegd, vraag voor UIG team.

uiteindelijk is er best ook consistente manier voor de ganse UIG code base

wimdetroyer commented on libs/map/src/lib/components/action/vl-map-click-action.wctest.ts:
```diff
@@ -0,0 +1,45 @@
+import {assert, aTimeout, fixture, html} from '@open-wc/testing';
+// TODO: zitten imports goed?
```
TODO

kspeltix committed:
fix: aanmaak van map files vermijden bij scss omzetting

de map files die gegenereerd worden zijn niet correct, dit geeft problemen bij afnemers

driesva commented on libs/map/src/lib/actions/map/map-with-actions.ts:
```diff
@@ -1,49 +1,46 @@
+
 export class VlMapWithActions extends Map {
-    private actions: any[];
+    private _actions: any[] = [];
```
Dit is een vraag voor UIG-team denk ik, het feit dat het private is. Is er dan nog een _ prefix nodig?

(naar backwards compatibility toe, geen idee of iemand in JS rechtstreeks dit zette)

driesva commented on libs/map/src/lib/components/action/vl-map-click-action-pindrop.ts:
```diff
@@ -0,0 +1,68 @@
+            height: 2rem;
+            border-radius: 2rem;
+            /* Set the marker color here */
+            background: #aa3300;
```
Hier heb je het rood overgenomen. Maar mss eerder blauw zoals Geopunt?
Aan UIG team om styling te bepalen zodat het cfr Vlaamse huisstijl is.

wimdetroyer replied:
heb er blauw van gemaakt. te bekijken @kspeltix

driesva commented on libs/map/src/lib/components/action/vl-map-click-action-pindrop.wctest.ts:
```diff
@@ -0,0 +1,17 @@
+
+
+    it('fixtures loads ok', async () => {
+        await pindropFixture();
```
detail: moet er geen assert komen hier?

wimdetroyer replied:
done

driesva commented on libs/map/src/lib/components/action/vl-map-click-action-pindrop.ts:
```diff
@@ -0,0 +1,68 @@
+ * VlMapClickActionPindrop
+ *
+ */
+@webComponent('vl-map-click-action-pindrop')
```
aangezien er Lit wordt gebruikt, geen ```@customElement```?

wimdetroyer replied:
Klopt, vergeten aan te passen.

driesva commented on libs/map/src/lib/components/action/vl-map-click-action.stories-doc.mdx:
````diff
@@ -0,0 +1,18 @@
+```html
+        <vl-map>
+            <vl-map-baselayer-grb-gray></vl-map-baselayer-grb-gray>
+            <vl-map-click-action onClick="${(click) => {onClick(click)}}"></vl-map-click-action>
````
`.onClick` ?

driesva commented on libs/map/src/lib/components/action/vl-map-click-action.ts:
```diff
@@ -0,0 +1,78 @@
+ */
+
+// TODO: crs doorsturen? vb: crs=EPSG:31370 => komt uit map??
+export interface ClickCallback {
```
ook een vraag voor UIG Team / Decibel team. OL API (Coordinate, ...) exposen of abstraheren naar eigen types?

wimdetroyer replied:
te bekijken na onderzoek getfeatureinfo call of dit niet onderhevig is aan verandering

driesva commented on libs/map/src/lib/components/action/vl-map-click-action.wctest.ts:
```diff
@@ -0,0 +1,45 @@
+
+    it('when the map is clicked, there is an overlay on the map with a pindrop element at the clicked location', async () => {
+        const map: VlMap = await clickActionFixture() as unknown as VlMap;
+        let pointerEvent = new PointerEvent("pointerdown", {
```
detail: mag ook const zijn vermoed ik, ook nog andere vars hieronder vermoed ik

driesva commented on libs/map/src/lib/components/action/vl-map-click-action.wctest.ts:
```diff
@@ -0,0 +1,45 @@
+
+        let overlays = map.map.getOverlays();
+
+        assert.isTrue(overlays.getLength() == 1);
```
muggenziften: `assert.equal` ?

driesva reviewed (commented):
Good job voor de eerste UIG bijdrage!

semantic-release-bot committed:
chore(release): 1.0.1 [skip ci]

## [1.0.1](https://github.com/milieuinfo/uigov-web-components/compare/v1.0.0...v1.0.1) (2023-01-31)

### Bug Fixes

* aanmaak van map files vermijden bij scss omzetting ([0b3084c](https://github.com/milieuinfo/uigov-web-components/commit/0b3084c99c171e910604cfa78736d7e2747c6beb))

wimdetroyer committed:
feat: KBOMG-27 - Eerste poging tot vl-click-action component

- gebruik maken van events ipv callback
- todos oplossen
- code wat polishen

wimdetroyer committed:
feat: KBOMG-27 - Eerste poging tot vl-click-action component

- laat VlMapClickedEvent bubblen

Commit:
feat: UIG-2287 - richtlijn Storybook en Cypress

Toegepast op Accordion en ProgressBar componenten als voorbeeld.

SBylemans committed:
feat: UIG-2301 - vl-header uitgebreid met simple mode

wimdetroyer committed:
feat: KBOMG-27 - Eerste poging tot vl-click-action component

- vergeten todo op te kuisen
